### DISCUSSION TITLE
Using Cognito providers with multiple user pools

### DISCUSSION THREAD

**Discussion by dbrxnds** [nextauthjs/next-auth#2540] [Help]
### Question 💬

Hi,

We are building an application in which we will be using Cognito for user management. As we are a SaaS solution we will be using multiple Cognito user pools as each customer gets their own. 

Is it possible with next-auth to set it up in such a way where you can dynamically load a poolId on the first mount?

We are currently using Amplify but it's a mess to work with and would prefer a neater solution.

Thank you all in advance!

### How to reproduce ☕️

N.A.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by munibrahman** [comment]
Hey, did you figure out if you were able to make this work?

It seems to me this isn't possible. Strictly speaking from experience I tried it with 2 userpools and it just ends up using a single user pool.
